carcara/src/checker/rules/strings.rs

```diff
@@ -323,6 +327,79 @@ fn extract_arguments(t: &Rc<Term>) -> Result<Vec<Rc<Term>>, CheckerError> {
     Ok(args_t.clone())
 }
 
+fn singleton_elim(pool: &mut dyn TermPool, r_list: Vec<Rc<Term>>) -> Rc<Term> {
+    match r_list.len() {
+        1 => r_list[0].clone(),
+        _ => pool.add(Term::Op(Operator::ReConcat, r_list)),
+    }
+}
+
+fn str_fixed_len_re(pool: &mut dyn TermPool, r: Rc<Term>) -> Result<usize, CheckerError> {
+    match r.as_ref() {
+        Term::Op(Operator::ReConcat, args) => {
+            if let [r_1, r_2 @ ..] = &args[..] {
+                let r_2_concat = concat(pool, r_2.to_vec());
+                Ok(str_fixed_len_re(pool, r_1.clone())? + str_fixed_len_re(pool, r_2_concat)?)
+            } else {
+                unreachable!()
+            }
```

Comment:
It is not correct, indeed :(

I confused this function with another one I previously implemented and ended up calling the wrong one. Also, I did not call `str_fixed_len_re` for each argument directly because I got too focused on how CPC represents `re.++` and `str.++` terms (`(str.++ a b c)` is transformed into `(str.++ a (str.++ b (str.++ c "")))`). However, with the representation in arrays that we have internally, it is, in fact, the same as applying `str_fixed_len_re` directly to each argument.

I am going to change that there and in other places where a similar behavior occurs.